apps/dashboard/app/controllers/application_controller.rb

```diff
@@ -17,11 +17,12 @@ def set_user
   end
 
   def set_nav_groups
+    groups = sys_app_groups + pinned_app_group
```

Comment:
This reminds me of AweSim and something I do not recall the design decision on.

We want to support different configurations for pinned apps. Such as:

1. all shared apps
2. shared apps from a specific user
3. individual apps specified
4. all apps by domain or some other metadata

I don't recall which of these are in scope for this issue or covered in another. But one problem will be that eventually the number of pinned apps may exceed the number of apps that we want to display in the dropdown, but may choose to display in the launch UI widget on the front page, by either:

1. setting a max height on the launch UI widget so that past 2 rows of icons you have to scroll to see more
2. just letting the user scroll, which will push the MOTD off the screen

However, I think that this is an edge case. For AweSim most end users will have only 1 or 2 icons. For class.osc.edu, there are more, but only like 5 total? And there the MOTD matters less. For OSC OnDemand we probably won't use pinned at the beginning - rather we will use favorites and recents that are built on top of the work here.

Do we have a plan for handling the dropdown getting too long?

Reply:
To caps - I'd guess this `pinned_app_group`/`@nav_groups` is the right place to catch filtering or limiting as it deals with the nav bar specifically - maybe cap through some configuration or use menus that pop up to the right? 

Recently used can clearly have a cap but I'm guessing other should have a configurable cap as well?

To other features, I'd say we should defer them to other tickets/PRs.

Reply:
I'm not sure if this location is the correct place for limiting, since we don't want to limit when we build the widget with the icons that display. But I'm not sure yet where it is appropriate. FWIW https://ruby-doc.org/core-3.0.0/Enumerable.html#method-i-take is an easy way to get the first few items of an array to iterate over.

Reply:
I would say for imposing limits on the navbar dropdowns, we can just push that to a separate issue to address later.